From 7c583a61ba1475adacfc7df0768e74407d211169 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 26 Jan 2018 13:52:53 +0800
Subject: [PATCH 3268/5242] MLK-17462-4: ARM64: dts : support TDM mode for
 audio board.

commit  3b6416f9bffc2991f343dac197b9039e5f45918a from
https://source.codeaurora.org/external/imx/linux-imx.git

Use a separate dts for tdm mode for ak5558.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8mq-evk-audio-tdm.dts     |   51 ++++++++++++++++++++
 2 files changed, 53 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-audio-tdm.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 19d9b9c..14b1609 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -59,4 +59,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-dcss-adv7535.dtb \
 				 fsl-imx8mq-evk-dcss-rm67191.dtb \
 				 fsl-imx8mq-evk-dual-display.dtb \
-				 fsl-imx8mq-evk-ak4497.dtb
+				 fsl-imx8mq-evk-ak4497.dtb \
+				 fsl-imx8mq-evk-audio-tdm.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-audio-tdm.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-audio-tdm.dts
new file mode 100644
index 0000000..772c77b
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-audio-tdm.dts
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+/ {
+	sound-ak4458 {
+		status = "okay";
+	};
+
+	sound-ak5558 {
+		fsl,tdm;
+		status = "okay";
+	};
+
+	sound-ak4497 {
+		status = "disabled";
+	};
+};
+
+&sai1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai1>;
+	assigned-clocks = <&clk IMX8MQ_CLK_SAI1_SRC>,
+			<&clk IMX8MQ_CLK_SAI1_DIV>;
+	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <0>, <49152000>;
+	status = "okay";
+};
+
+&sai5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai5>;
+	assigned-clocks = <&clk IMX8MQ_CLK_SAI5_SRC>,
+			<&clk IMX8MQ_CLK_SAI5_DIV>;
+	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <0>, <49152000>;
+	fsl,sai-asynchronous;
+	status = "okay";
+};
-- 
1.7.9.5

